connectors/bigtable/flink-connector-gcp-bigtable/src/test/java/com/google/flink/connector/gcp/bigtable/table/BigtableTableTest.java

```diff
@@ -0,0 +1,252 @@
+
+        ResolvedSchema schema = getResolvedSchema(true);
+
+        BigtableDynamicTableSink sink =
+                (BigtableDynamicTableSink) FactoryMocks.createTableSink(schema, options);
+
+        ReadableConfig connectorOptions = sink.connectorOptions;
+        assertNull(connectorOptions.get(BigtableConnectorOptions.COLUMN_FAMILY));
+        assertTrue(connectorOptions.get(BigtableConnectorOptions.USE_NESTED_ROWS_MODE));
+        assertEquals(connectorOptions.get(BigtableConnectorOptions.TABLE), TestingUtils.TABLE);
+        assertEquals(
+                connectorOptions.get(BigtableConnectorOptions.INSTANCE), TestingUtils.INSTANCE);
+        assertEquals(connectorOptions.get(BigtableConnectorOptions.PROJECT), TestingUtils.PROJECT);
+
+        assertEquals(sink.rowKeyField, TestingUtils.ROW_KEY_FIELD);
+        assertEquals(sink.resolvedSchema, schema);
+        assertNull(sink.parallelism);
+    }
+
+    @Test
+    public void testCopy() {
+        ResolvedSchema schema = getResolvedSchema(true);
+
+        Map<String, String> options = getRequiredOptions();
+        options.put(BigtableConnectorOptions.USE_NESTED_ROWS_MODE.key(), "true");
+
+        BigtableDynamicTableSink sink =
+                (BigtableDynamicTableSink) FactoryMocks.createTableSink(schema, options);
+        BigtableDynamicTableSink sink2 =
+                (BigtableDynamicTableSink) FactoryMocks.createTableSink(schema, options);
+
+        assertEquals(sink2, sink.copy());
+        assertTrue(sink2.equals(sink.copy()));
+    }
+
+    @Test
+    public void testCopyParallelism() {
+        ResolvedSchema schema = getResolvedSchema(false);
+
+        Map<String, String> options = getRequiredOptions();
+        options.put(BigtableConnectorOptions.COLUMN_FAMILY.key(), TestingUtils.COLUMN_FAMILY);
+        options.put(BigtableConnectorOptions.SINK_PARALLELISM.key(), "2");
+
+        BigtableDynamicTableSink sink =
+                (BigtableDynamicTableSink) FactoryMocks.createTableSink(schema, options);
+        BigtableDynamicTableSink sink2 =
+                (BigtableDynamicTableSink) FactoryMocks.createTableSink(schema, options);
+
+        assertEquals(sink2, sink.copy());
+        assertTrue(sink2.equals(sink.copy()));
+    }
+
+    @Test
+    public void testSummaryString() {
+        ResolvedSchema schema = getResolvedSchema(false);
+
+        String parallelism = "2";
+        Map<String, String> options = getRequiredOptions();
+        options.put(BigtableConnectorOptions.USE_NESTED_ROWS_MODE.key(), "true");
+        options.put(BigtableConnectorOptions.SINK_PARALLELISM.key(), parallelism);
+
+        BigtableDynamicTableSink sink =
+                (BigtableDynamicTableSink) FactoryMocks.createTableSink(schema, options);
+
+        String summaryString =
+                String.format(
+                        "BigtableSink("
+                                + "parallelism=%s, "
```

Comment:
doesnt need to be a test but how does string formatting work if parallelism is null?

Reply:
Sure, I thought it was too much hahaha. Added